network-core/src/server/block.rs

```diff
@@ -111,8 +114,8 @@ pub trait BlockService: P2pService {
     /// to the server's tip.
     fn pull_blocks_to_tip(&mut self, from: &[Self::BlockId]) -> Self::PullBlocksToTipFuture;
 
-    /// Get block headers, walking forward in a range between any of the given
-    /// starting points, and the ending point.
+    /// Get block headers, walking the chain forward in a range between any
+    /// of the given starting points, and the ending point.
```

Comment:
Maybe this should be more specific about which starting point is used, e.g. the closest ancestor of the ending point that the server has available. Also, what happens if none of the starting points is known?

Reply:
If the filtered checkpoint list is empty, an empty stream is returned. In the push direction, we make a bogus `PushHeaders` call with an empty stream, too. Perhaps I should change this to send a `NotFound` error instead, good point.